Library/Homebrew/cask/installer.rb

```diff
@@ -571,6 +574,107 @@ def purge_caskroom_path
       gain_permissions_remove(@cask.caskroom_path)
     end
 
+    sig { void }
+    def forbidden_tap_check
+      forbidden_taps = Homebrew::EnvConfig.forbidden_taps
+      return if forbidden_taps.blank?
+
+      forbidden_taps_set = Set.new(forbidden_taps.split.filter_map do |tap|
+        Tap.fetch(tap)
+      rescue Tap::InvalidNameError
+        opoo "Invalid tap name in `HOMEBREW_FORBIDDEN_TAPS`: #{tap}"
+        nil
+      end)
```

Comment:
It would also be possible to have method that takes a list of casks or formulae and returns the first invalid one by tap, cask, formula or license.

Reply:
> In terms of logic that could be shared between casks and formulae right now, this bit could be moved to a `Tap.forbidden` method or something similar.

Could be but feels overkill on Rule Of 3: https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

> It would also be possible to have method that takes a list of casks or formulae and returns the first invalid one by tap, cask, formula or license.

It would be possible but I'm not convinced it would be more readable.